### PR TITLE
Add edit applications

### LINKED ISSUE #1153: Implement My Profile - Integrations - My Applications - add / edit

### PR DESCRIPTION
**Pull request by eatonphil**
Builds on #1265, closes #1153

screen shot 2017-02-23 at 10 09 57 am

screen shot 2017-02-23 at 10 10 30 am

### CODE REVIEW

File: src/api/apiResultActionReducerGenerator.js [hunk 1]
```diff
@@ -30,13 +30,22 @@ function fullyQualified(resource) {
   return path;
 }
 
+function parseIntIfActualInt(string) {
```
Comment (mparke) [inline src/api/apiResultActionReducerGenerator.js]: if this is trying to do what I think it is,

`isNaN(string)` can be used to check the same thing

Comment (eatonphil) [inline src/api/apiResultActionReducerGenerator.js]: kk, thanks!

**Comment by abemassry** [comment]
got this when testing edit filled out the fields and added a thumbnail:
1. modal didn't close after Save button was pressed.  This may be due to changes in master and once it's rebased it will get fixed.
1. PUT https://api.alpha.linode.com/v4/account/clients/d99df0ca4de384959306/thumbnail 413 (Request Entity Too Large) was unhandled (not sure if this is in the scope of the PR)
1. Fetch API cannot load https://api.alpha.linode.com/v4/account/clients/d99df0ca4de384959306/thumbnail. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access. The response had HTTP status code 413. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
1. CreateApplication.js?5a57:43  TypeError: Failed to fetch
1. Uncaught (in promise) TypeError: response.json is not a function

File: src/profile/integrations/components/EditApplication.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      errors: {},
+      thumbnail: '',
+      name,
+      redirect,
+    };
+  }
+
+  submitAction = async (name, redirect) => {
+    const { id } = this.props;
+    await this.props.dispatch(clients.put({ name, redirect_uri: redirect }, [id]));
```
Comment (mparke) [inline src/profile/integrations/components/EditApplication.js]: can we pass this fn on edit to save from an additional class extend?

**Comment by mparke** [comment]
should we name or show an existing thumbnail on edit? 

also, is there an issue for thumnail 401s?

screen shot 2017-03-02 at 10 52 19 am

**Comment by eatonphil** [comment]
> should we name or show an existing thumbnail on edit?

Maybe as a feature request? I don't feel too strongly either way and it's quite clear from the previous screen whether or not there is a thumbnail.

> also, is there an issue for thumnail 401s?

A known issue that the API is dealing with. They are dropping permissions required so we can reference the thumbnails in img tags (like we are doing here).

**Comment by eatonphil** [comment]
thank you!

**Pull request merged**